File: Java Script - Introduction to Programming/5.1. Loops/11. Even Position .js
function oddEvenPos(arg) {
  let loop = Number(arg[0])
  let [oddSum, oddMin, oddMax] = [0, Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER]
  let [evenSum, evenMin, evenMax] = [0, Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER]

  for (let i = 1; i <= loop ;i++) {
    let num = Number(arg[i])
    if (i & 1) {
      oddSum += num
      if (num > oddMax) oddMax = num
      if (num < oddMin) oddMin = num

    } else {
      evenSum += num
      if (num > evenMax) evenMax = num
      if (num < evenMin) evenMin = num
    }
  }

  if (oddMin === Number.MAX_SAFE_INTEGER) oddMin = 'No'
  if (evenMin === Number.MAX_SAFE_INTEGER) evenMin = 'No'
  if (oddMax === Number.MIN_SAFE_INTEGER) oddMax = 'No'
  if (evenMax === Number.MIN_SAFE_INTEGER) evenMax = 'No'


  return `OddSum = ${oddSum}\nOddMin = ${oddMin}\nOddMax = ${oddMax}\nEvenSum = ${evenSum}\nEvenMin = ${evenMin}\nEvenMax = ${evenMax}`
}

// console.log(oddEvenPos([6, 2, 3, 5, 4, 2, 1]))
// console.log(oddEvenPos([2, 1.5, -2.5]))




//----------------------------------------(2)---------------------------------WTF ????
// function oddEvenPosition(arg) {       
//   arg.shift()
//   arg = arg.map(Number)
//   let evenPositions = arg.filter((el, index) => index % 2 === 0)
//   let oddPositions = arg.filter((el, index) => index & 1)

//   function result(arr, str) {
//     sum = arr.reduce((a, b) => a + b)
//     min = Math.min(...arr)
//     max = Math.max(...arr)
//     return `${str}Sum=${sum},\n${str}Min=${min},\n${str}Max=${max},`
//   }

//   if (!oddPositions.length)
//     return `${result(evenPositions, 'Odd')}\nOddSum=0,\nOddMin=No,\nOddMax=No,`
//   else if (!evenPositions.length)
//     return `${result(oddPositions, 'Even')}\nEvenSum=0,\nEvenMin=No,\nEvenMax=No,`
//   else
//     return `${result(evenPositions, 'Odd')}\n${result(oddPositions, 'Even')}`
// }


//console.log(oddEvenPosition(['6', '2', '3', '5', '4', '2', '1']))
//console.log(oddEvenPosition(['1', '1']))
// console.log(oddEvenPosition(['1', '-5']))
//console.log(oddEvenPosition(['5', '3', '-2', '8', '11', '-3']))
//oddEvenPosition(['0'])

File: Pyrhon - Introduction to Programming/5.1. Loops/11. Even Position .py
import  sys

n = int(input())
odd_sum = 0
odd_min = sys.maxsize
odd_max = -sys.maxsize
even_sum = 0
even_min = sys.maxsize
even_max = -sys.maxsize

for i in range(0, n):
    current_number = input()
    if '.' in current_number:
      current_number = float(current_number)
    else:
      current_number = int(current_number)

    if i % 2 == 1:
        even_sum += current_number
        if current_number < even_min:
            even_min = current_number
        if current_number > even_max:
            even_max = current_number
    else:
        odd_sum += current_number
        if current_number < odd_min:
            odd_min = current_number
        if current_number > odd_max:
            odd_max = current_number


if isinstance(odd_sum, float) and str(odd_sum)[-1:] == '0':
  odd_sum = int(odd_sum)

print(f"OddSum={odd_sum},")

if odd_min == sys.maxsize:
    print(f"OddMin=No,")
    print(f"OddMax=No,")
else:
     print(f"OddMin={odd_min},")
     print(f"OddMax={odd_max},")

if  isinstance(even_sum, float) and str(even_sum)[-1:] == '0':
  even_sum = int(even_sum)

print(f"EvenSum={even_sum},")

if even_min == sys.maxsize:
    print(f"EvenMin=No,")
    print(f"EvenMax=No")
else:
    print(f"EvenMin={even_min},")
    print(f"EvenMax={even_max}")



'''
4
1.5
1.75
1.5
1.75
'''



